### PR TITLE
[`pylint`] invalid-characters-*

### PR DESCRIPTION
**Pull request by r3m0t**

### CODE REVIEW

File: crates/ruff/src/settings/rule_table.rs [hunk 1]
```diff
@@ -26,6 +26,11 @@ impl RuleTable {
         self.enabled.contains_key(code)
     }
 
+    /// Returns whether any of the given rules should be checked.
+    pub fn any_enabled(&self, codes: &[&Rule]) -> bool {
+        codes.iter().any(|c| self.enabled.contains_key(c))
+    }
```
Comment (MichaReiser) [inline crates/ruff/src/settings/rule_table.rs:32]: Neat! 

We can improve the ergonomics (reduce the amount of `&` a caller must write) by: 

1. Change `Rule` to implement `Copy` (Rule is only 2 bytes large, which is cheaper than passing a reference that is 8 bytes) by changing the derive macro [here](https://github.com/charliermarsh/ruff/blob/aa51ecedc5ebbba5173350d45f0ea4ae9ae41df6/crates/ruff_macros/src/register_rules.rs#L32-L45)
2. Accept any iterable 
```suggestion
    pub fn any_enabled(&self, codes: impl IntoIterator<Item=Rule>) -> bool {
        codes.into_iter().any(|c| self.enabled.contains_key(c))
    }
```


You can then use it like this

```rust
let enforce_ambiguous_unicode_character = settings.rules.any_enabled([
        Rule::AmbiguousUnicodeCharacterString,
        Rule::AmbiguousUnicodeCharacterDocstring,
        Rule::AmbiguousUnicodeCharacterComment,
    ]);
```

File: crates/ruff/src/checkers/tokens.rs [hunk 2]
```diff
@@ -23,39 +23,43 @@ pub fn check_tokens(
...
-            .rules
-            .enabled(&Rule::MultipleStatementsOnOneLineSemicolon)
-        || settings.rules.enabled(&Rule::UselessSemicolon);
+    let enforce_compound_statements = settings.rules.any_enabled(&[
+        &Rule::MultipleStatementsOnOneLineColon,
+        &Rule::MultipleStatementsOnOneLineSemicolon,
+        &Rule::UselessSemicolon,
+    ]);
     let enforce_invalid_escape_sequence = settings.rules.enabled(&Rule::InvalidEscapeSequence);
     let enforce_implicit_string_concatenation = settings
```
Comment (MichaReiser) [inline crates/ruff/src/checkers/tokens.rs:51]: Nit: Use `any_enabled`?

File: crates/ruff/src/rules/pylint/rules/invalid_string_characters.rs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+/// ## Example
+/// ```python
+/// x = ''
+/// ```
+///
+/// Use instead:
+/// ```python
+/// x = '\x1B'
+/// ```
+
```
Comment (MichaReiser) [inline crates/ruff/src/rules/pylint/rules/invalid_string_characters.rs:88]: We prefer not to have an empty line between the doc comment and a type's attributes.

```suggestion
```

File: crates/ruff/src/rules/pylint/rules/invalid_string_characters.rs [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+/// ## Example
+/// ```python
+/// x = ''
+/// ```
+///
+/// Use instead:
+/// ```python
+/// x = '\0'
+/// ```
+
```
Comment (MichaReiser) [inline crates/ruff/src/rules/pylint/rules/invalid_string_characters.rs:119]: We prefer not to have an empty line between the doc comment and a type's attributes.

```suggestion
```

**Review by MichaReiser — APPROVED** [review]

**Comment by charliermarsh** [comment]
Sorry for the delay, will get to this tomorrow.

**Pull request merged**